qiskit_experiments/framework/experiment_data.py

```diff
@@ -69,14 +70,15 @@
 from qiskit_experiments.database_service.utils import objs_to_zip, zip_to_objs
 
 from .containers.figure_data import FigureData, FigureType
+from .provider_interfaces import Job, Provider
```

Comment:
I wasn't sure if it was convenient to use the `Job` and `Provider` names here (they basically mean what you expect them to mean) or confusing (they don't quite mean exactly what you might first assume they mean). Maybe `JobLike` and `ProviderLike` would be better or something like that?